### PR TITLE
Fixing and improving few wcag aspects

### PR DESCRIPTION
**Pull request by Vishwaas**
With regards to this [issue](https://github.com/amsul/pickadate.js/issues/917)
Pending
 - [x] Update information on changes
 - [x] Verify changes

### CODE REVIEW

File: lib/picker.date.js [hunk 1]
```diff
@@ -1031,7 +1031,7 @@ DatePicker.prototype.nodes = function( isOpen ) {
                     ( !next && viewsetObject.year <= minLimitObject.year && viewsetObject.month <= minLimitObject.month ) ?
                     ' ' + settings.klass.navDisabled : ''
                 ),
-                'data-nav=' + ( next || -1 ) + ' ' +
+                'data-nav=' + ( next || -1 ) + ' ' + 'tabindex="0" ' +
```
Comment (Vishwaas) [inline lib/picker.date.js]: making the next and previous month navigation button keyboard accessible

File: lib/picker.date.js [hunk 2]
```diff
@@ -1222,16 +1222,15 @@ DatePicker.prototype.nodes = function( isOpen ) {
 
                                             return klasses.join( ' ' )
                                         })([ settings.klass.day ]),
-                                        'data-pick=' + targetDate.pick + ' ' + _.ariaAttr({
+                                        'data-pick=' + targetDate.pick + ' ' + 'id=' + targetDate.pick + ' tabindex="0" ' + _.ariaAttr({
```
Comment (Vishwaas) [inline lib/picker.date.js]: adding id to element to support activedescendent
making each date keyboard accessible

File: lib/picker.date.js [hunk 3]
```diff
@@ -1222,16 +1222,15 @@ DatePicker.prototype.nodes = function( isOpen ) {
...
                                             role: 'gridcell',
                                             label: formattedDate,
                                             selected: isSelected && calendar.$node.val() === formattedDate ? true : null,
-                                            activedescendant: isHighlighted ? true : null,
+                                            activedescendant: isHighlighted ? targetDate.pick : null,
```
Comment (Vishwaas) [inline lib/picker.date.js]: activedescendent should not be a boolean but the id of the element.
element here is each date of the month

File: lib/picker.js [hunk 4]
```diff
@@ -233,7 +233,6 @@ function PickerConstructor( ELEMENT, NAME, COMPONENT, OPTIONS ) {
 
                 // Add the “active” class.
                 $ELEMENT.addClass( CLASSES.active )
-                aria( ELEMENT, 'expanded', true )
```
Comment (Vishwaas) [inline lib/picker.js]: jaws throws out an error if you have expanded on the trigger element, it should be used with the correct role.

File: lib/picker.date.js [hunk 5]
```diff
@@ -1222,16 +1222,15 @@ DatePicker.prototype.nodes = function( isOpen ) {
...
                                             disabled: isDisabled ? true : null
                                         })
                                     ),
-                                    '',
-                                    _.ariaAttr({ role: 'presentation' })
+                                    ''
```
Comment (Vishwaas) [inline lib/picker.date.js]: having role presentation does not read the date

**Comment by Vishwaas** [comment]
@amsul Ready for review, we were facing certain wcag issues with the component. Hence addressing the same here

File: lib/picker.date.js (hunk 2 above)
Comment (DanielRuf) [reply lib/picker.date.js]: Not sure if this works well with multiple datepickers (duplicate IDs). Do we have some other generated instance string that we can append to the id?

**Comment by Vishwaas** [comment]
@DanielRuf  changes made

**Comment by Vishwaas** [comment]
@DanielRuf i had to change to the reference for ID to be more explicit take a look at the commit "pass id"

**Comment by Vishwaas** [comment]
@DanielRuf is this still awaiting reply?

File: lib/picker.date.js (hunk 3 above)
Comment (smrhoney) [reply lib/picker.date.js]: Same problem is in picker.time.js as well. However, the supposed fix is being added to the wrong role. 
Reference: https://www.w3.org/TR/wai-aria-1.1
Activedescendant needs to be added to roles grid and combobox for DatePicker and added for roles combobox and listbox for TimePicker.  

Each grid item and list item needs an id attribute set to be passed to aria-activedescendant attribute.

**Comment by amsul** [comment]
@Vishwaas are you on [Spectrum](https://spectrum.chat)? I'd love to have a chat with you regarding WCAG in v5.

**Comment by amsul** [comment]
@Vishwaas didn't see anything on Spectrum yet..

**Comment by Vishwaas** [comment]
@amsul   I did ping couple of times, do you want to schedule a particular time? Or an email I can reach out to?

**Comment by amsul** [comment]
@Vishwaas merging this in! Thanks for the contribution 🙏 

Hopefully it resolves all the issues outlined in https://github.com/amsul/pickadate.js/issues/917

**Comment by Vishwaas** [comment]
@amsul  thank you. I have another PR coming up improving a bit more on the accessibility perspective, however this should address most of those mentioned in #917

**Pull request merged**